### PR TITLE
Enable to connect to the existed ETCD cluster in GraphScope

### PR DESCRIPTION
**Pull request by wuyueandrew**
## What do these changes do?

add new start param etcd_addrs to specify etcd ,such as        

```
     --k8s_etcd_addrs
     "10.106.250.244:2379,"
```

## Related issue number

Fixes https://github.com/alibaba/GraphScope/issues/1343

### CODE REVIEW

File: coordinator/gscoordinator/cluster.py [hunk 1]
```diff
@@ -908,10 +899,33 @@ def _create_interactive_engine_service(self):
...
+
+    def _get_etcd_endpoints(self):
+        etcd_addrs = []
+        if not self._etcd_addrs:
+            port = self._random_etcd_listen_client_service_port
+            etcd_addrs.append("%s:%s" % (self._etcd_service_name, port))
+            for i in range(self._etcd_num_pods):
+                etcd_addrs.append("%s-%d:%s" % (self._etcd_name, i, port))
+        else:
+            for i in self._etcd_addrs.split(sep=","):
```
Comment (siyuan0322) [inline coordinator/gscoordinator/cluster.py:919]: You can just use  `etcd_addrs = self._etcd_addrs.split(",")` to simplify the code.

File: coordinator/gscoordinator/cluster.py [hunk 2]
```diff
@@ -908,10 +899,33 @@ def _create_interactive_engine_service(self):
...
+                etcd_addrs.append(i)
+        etcd_endpoints = []
+        for i in etcd_addrs:
+            if len(i) != 0:
+                etcd_endpoints.append("http://%s" % (i))
```
Comment (siyuan0322) [inline coordinator/gscoordinator/cluster.py:924]: To simplify the code, you can use: `etcd_endpoints = ["http://%s" % i for i in etcd_addrs if i]`

File: coordinator/gscoordinator/coordinator.py [hunk 3]
```diff
@@ -1297,6 +1297,12 @@ def parse_sys_args():
         default="256Mi",
         help="Memory of engine container, suffix with ['Mi', 'Gi', 'Ti'].",
     )
+    parser.add_argument(
+        "--k8s_etcd_addrs",
+        type=str,
+        default=None,
+        help="The addr of etcd cluster, format such as 'etcd01:port,etcd02:port,etcd03:port' ",
```
Comment (siyuan0322) [inline coordinator/gscoordinator/coordinator.py:1304]: The addr of external etcd cluster, with formats like `xxx`.

**Review by siyuan0322 — COMMENTED** [review]
The logic looks fine, but there should be a parameter in the `Session` for user to pass the addr.

**Review by lidongze0629 — COMMENTED** [review]
Thanks for submitting your pull request, it's a great job!

And I think there are still some work to be done before merged into main branch.

1. In addition to modifying the parts of k8s, the way launching session locally also needs to be changed. See detail from [create etcd locally](https://github.com/alibaba/GraphScope/blob/main/coordinator/gscoordinator/launcher.py#L313)

2. And at the same time, you also need to update the code of client to ensure `etcd_addrs` can be passed to the coordinator, some useful info for you, such as [session defination](https://github.com/alibaba/GraphScope/blob/main/python/graphscope/client/session.py#L253), [launch coordinator from client](https://github.com/alibaba/GraphScope/tree/main/python/graphscope/deploy)

File: coordinator/gscoordinator/coordinator.py (hunk 3 above)
Comment (wuyueandrew) [reply coordinator/gscoordinator/coordinator.py:1304]: thanks for these suggestions.

**Comment by wuyueandrew** [comment]
thanks, i'll fix these points

**Review by sighingnow — APPROVED** [review]

**Review by siyuan0322 — APPROVED** [review]

**Review by lidongze0629 — APPROVED** [review]
LGTM.

File: python/graphscope/deploy/hosts/cluster.py [hunk 4]
```diff
@@ -93,6 +95,8 @@ def _launch_coordinator(self):
             "{}".format(str(self._num_workers)),
             "--hosts",
             "{}".format(",".join(self._hosts)),
+            "--etcd_addrs",
+            self._etcd_addrs,
```
[the remark this answers is not in the document]
Comment (wuyueandrew) [reply python/graphscope/deploy/hosts/cluster.py]: thanks, i'll fix it

**Pull request merged**